### ISSUE TITLE
Checkbox without label doesn't render nicely in a list

### ISSUE THREAD

**Issue by stefanloerwald** [Material-Blazor/Material.Blazor#215]
With the code

```
<ul>
    <li>
        <BlazorMdc.MTCheckbox />Hello text
    </li>
</ul>
```

the rendered output is

![image](https://user-images.githubusercontent.com/10850250/90140061-7f047000-dd79-11ea-8b63-37b329d0bd96.png)

The issue seems to be the outer div of the Checkbox, see third line:

```html
<ul><!--!-->
    <li><!--!-->
        <!--!--><div class="" _bl_0135c0ac-b66a-4579-a8bb-dff1620943e5=""><!--!-->
    <div class="mdc-touch-target-wrapper"><!--!-->
        <div class="mdc-checkbox mdc-checkbox--touch mdc-checkbox--upgraded mdc-ripple-upgraded mdc-ripple-upgraded--unbounded" _bl_4e4b3c67-10f3-425a-a595-4ed3fc4456ba="" style="--mdc-ripple-fg-size:24px; --mdc-ripple-fg-scale:1.66667; --mdc-ripple-left:8px; --mdc-ripple-top:8px;"><!--!-->

            <input type="checkbox" class="mdc-checkbox__native-control " id="bmdc_a5220989-d9e4-4718-a869-0a1844a5859a"><!--!-->
            <!--!--><div class="mdc-checkbox__background">
                <svg class="mdc-checkbox__checkmark" viewBox="0 0 24 24">
                    <path class="mdc-checkbox__checkmark-path" fill="none" d="M1.73,12.91 8.1,19.28 22.79,4.59"></path>
                </svg>
                <div class="mdc-checkbox__mixedmark"></div>
            </div>
            <div class="mdc-checkbox__ripple"></div>
        </div><!--!-->
    </div><!--!-->
    <label for="bmdc_a5220989-d9e4-4718-a869-0a1844a5859a"></label><!--!-->
</div><!--!-->Hello text
    </li><!--!-->
</ul>
```

This line is only relevant if the checkbox is inside a form field, as far as I can see from the code:

```
<div @ref="@FormReference" class="@FormFieldClass">
```

I'd propose to make that div optional:

```
@code {
    // wrapping div if it is in a form
    private RenderFragment<RenderFragment> with_form => (ChildContent) =>@<div @ref="@FormReference" class="@FormFieldClass">@ChildContent</div>;
    // no wrapping div if it is not in a form
    private RenderFragment<RenderFragment> no_form => (ChildContent) => @<text>@ChildContent</text>;
}

@{
    // decide which wrapper to take
    var wrapper = (IsFormField || !(string.IsNullOrWhiteSpace(Label))) ? with_form : no_form;
    // pass the content to it
    @wrapper(@<text>
        <div class="mdc-touch-target-wrapper">
            <div @ref="@ElementReference"
                 @attributes="@AttributesToSplat(SplatType.ExcludeHtmlExcludingClassAndStyle)">

                <input @bind="@CheckedValue"
                       type="checkbox"
                       class="mdc-checkbox__native-control @FieldClass"
                       id="@Id"
                       @attributes="AttributesToSplat(SplatType.HtmlExcludingClassAndStyle)" />
                <div class="mdc-checkbox__background">
                    <svg class="mdc-checkbox__checkmark"
                         viewBox="0 0 24 24">
                        <path class="mdc-checkbox__checkmark-path"
                              fill="none"
                              d="M1.73,12.91 8.1,19.28 22.79,4.59" />
                    </svg>
                    <div class="mdc-checkbox__mixedmark"></div>
                </div>
                <div class="mdc-checkbox__ripple"></div>
            </div>
        </div>
        <label for="@Id">@Label</label>
    </text>)
}
```

**Comment by stefanloerwald** [comment]
I should probably say: the code above only compiles, but is otherwise untested ;-)

**Comment by simonziegler** [comment]
@stefanloerwald, I'm slightly struggling to get this right. Any chance you could submit a PR?

**Comment by stefanloerwald** [comment]
This is resolved with version 1.0.0-Preview.13. Thank you!
